codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanDecoder.java

```diff
@@ -213,4 +138,114 @@ private static void insert(Node root, int symbol, int code, byte length) {
             current.children[i] = terminal;
         }
     }
+
+    private static final class DecoderProcessor implements ByteProcessor {
+        // TODO: Is this a good size ?
+        private static final int INITIAL_SIZE = 64;
+        private byte[] bytes = new byte[INITIAL_SIZE];
+        private int index;
+
+        private Node node = ROOT;
+        private int current;
+        private int currentBits;
+        private int symbolBits;
+
+        void reset() {
+            node = ROOT;
+            current = 0;
+            currentBits = 0;
+            symbolBits = 0;
+            bytes = new byte[INITIAL_SIZE];
```

Comment:
@normanmaurer interesting idea ... would the Nodes then have to store offset/length information into the bytes array? I wonder if this would become a maintenance burden as strings come and go?

Continuing with your current solution, perhaps we could produce less garbage by keeping the bytes array around as a scratch buffer and then copying `AsciiString` objects out of it.  That way, there should be overall fewer byte[] allocations since the scratch buffer growth will effectively stop over time.